.github/workflows/ci.yaml

```
@@ -313,3 +313,40 @@ jobs:
         if: ${{ steps.filter-images.outputs.kube-webhook-certgen == 'true' }}
         run: |
           cd images/kube-webhook-certgen && make test test-e2e
+
+  rancher-validate: 
+    runs-on: ubuntu-latest
+    env:
+      PLATFORMS: linux/amd64
+    container:
+      image: rancher/dapper:v0.5.3
+      env:
+        GIT_IN_DAPPER: true
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: validate
+        run: |
+          dapper validate
+
+  rancher-build:
+    needs: rancher-validate
+    runs-on: ubuntu-latest
+    env:
+      PLATFORMS: linux/amd64
+    container:
+      image: rancher/dapper:v0.5.3
+      env:
+        GIT_IN_DAPPER: true
+    steps:
+      - name: Install git
+        run: |
+          apk add git
```

```suggestion
          apk add --no-cache git
```

Also, do you see the job fails if you do not install git here? Is git not builtin?

---

No git is not built-in in both the ubuntu and rancher/dapper images. If git is not installed, the checkout step is downloading the repository using GitHub REST API and while running any git command in the build job, it's failing, and throwing an error saying `fatal: not a git repository (or any of the parent directories): .git`. So before checkout we need to install git. you can refer to the checkout logs here https://github.com/rancher/ingress-nginx/runs/5184047717?check_suite_focus=true